src/transformers/modeling_flax_utils.py

```diff
@@ -111,13 +112,33 @@ def __init__(
         # Those are public as their type is generic to every derived classes.
         self.key = PRNGKey(seed)
         self.dtype = dtype
+        self.input_shape = input_shape
 
-        # randomly initialized parameters
-        random_params = self.init_weights(self.key, input_shape)
+        # To check if the model was intialized automatically.
+        self._is_intialized = do_init
+
+        if do_init:
+            # randomly initialized parameters
+            random_params = self.init_weights(self.key, input_shape)
+            params_shape_tree = jax.eval_shape(lambda params: params, random_params)
+        else:
+            init_fn = partial(self.init_weights, input_shape=input_shape)
+            params_shape_tree = jax.eval_shape(init_fn, self.key)
+
+            logger.warning(
```

Comment:
```suggestion
            logger.info(
```

Think a warning is a bit strong here no as this is supposedly intentional